src/components/clusterWizard/BaremetalInventory.tsx

```diff
@@ -8,36 +9,22 @@ import {
   Text,
   TextContent,
 } from '@patternfly/react-core';
-
 import PageSection from '../ui/PageSection';
 import HostsTable from './HostsTable';
 import ClusterWizardToolbar from './ClusterWizardToolbar';
 import { ToolbarButton, ToolbarText } from '../ui/Toolbar';
-import { Cluster } from '../../api/types';
 import { Link } from 'react-router-dom';
 import { WizardStep } from '../../types/wizard';
-import { getClusterHosts } from '../../api/clusters';
-import useApi from '../../api/useApi';
 import { ResourceUIState } from '../../types';
 import { DiscoveryImageModalButton } from './discoveryImageModal';
+import { selectCurrentClusterState } from '../../selectors/currentCluster';
 
 interface BareMetalInventoryProps {
-  cluster: Cluster;
   setStep: React.Dispatch<React.SetStateAction<WizardStep>>;
 }
 
-const BaremetalInventory: React.FC<BareMetalInventoryProps> = ({ cluster, setStep }) => {
-  const [{ data: hosts, uiState }, fetchHosts] = useApi(getClusterHosts, cluster.id, {
-    manual: true,
-    initialUIState: cluster.hosts?.length ? ResourceUIState.LOADED : ResourceUIState.EMPTY,
-  });
-
-  React.useEffect(() => {
-    const timer = setTimeout(() => {
-      fetchHosts();
-    }, 5000);
-    return () => clearTimeout(timer);
-  });
+const BaremetalInventory: React.FC<BareMetalInventoryProps> = ({ setStep }) => {
+  const { data: cluster, uiState } = useSelector(selectCurrentClusterState);
```

Comment:
with keeping the cluster prop we can just get the uiState using `useSelector(selectCurrentClusterUIState)`